presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayAbstractMinMaxFunction.java

```diff
@@ -0,0 +1,187 @@
+{
+    private final OperatorType operatorType;
+    private final String functionName;
+    private final Signature signature;
+
+    private static final Map<Class<?>, MethodHandle> METHOD_HANDLES = ImmutableMap.<Class<?>, MethodHandle>builder()
+            .put(boolean.class, methodHandle(ArrayAbstractMinMaxFunction.class, "booleanArrayMinMax", MethodHandle.class, Type.class, Slice.class))
+            .put(long.class, methodHandle(ArrayAbstractMinMaxFunction.class, "longArrayMinMax", MethodHandle.class, Type.class, Slice.class))
+            .put(double.class, methodHandle(ArrayAbstractMinMaxFunction.class, "doubleArrayMinMax", MethodHandle.class, Type.class, Slice.class))
+            .put(Slice.class, methodHandle(ArrayAbstractMinMaxFunction.class, "sliceArrayMinMax", MethodHandle.class, Type.class, Slice.class))
+            .put(void.class, methodHandle(ArrayAbstractMinMaxFunction.class, "arrayWithUnknownType", MethodHandle.class, Type.class, Slice.class))
+            .build();
+
+    public ArrayAbstractMinMaxFunction(OperatorType operatorType, String functionName)
+    {
+        this.operatorType = operatorType;
+        this.functionName = functionName;
+        this.signature = new Signature(functionName, ImmutableList.of(comparableTypeParameter("E")), "E", ImmutableList.of("array<E>"), false, false);
+    }
+
+    @Override
+    public Signature getSignature()
+    {
+        return signature;
+    }
+
+    @Override
+    public boolean isHidden()
+    {
+        return false;
+    }
+
+    @Override
+    public boolean isDeterministic()
+    {
+        return true;
+    }
+
+    @Override
+    public String getDescription()
+    {
+        return "Get min value of array";
+    }
+
+    @Override
+    public FunctionInfo specialize(Map<String, Type> types, int arity, TypeManager typeManager, FunctionRegistry functionRegistry)
+    {
+        checkArgument(types.size() == 1, "Expected one type, got %s", types);
+        Type elementType = types.get("E");
+        checkArgument(elementType.isOrderable(), "Type must be orderable");
+
+        MethodHandle compareMethodHandle = functionRegistry.resolveOperator(operatorType, ImmutableList.of(elementType, elementType)).getMethodHandle();
+        MethodHandle methodHandle = METHOD_HANDLES.get(elementType.getJavaType()).bindTo(compareMethodHandle).bindTo(elementType);
+        requireNonNull(methodHandle, "methodHandle is null");
+
+        Signature signature = new Signature(functionName, elementType.getTypeSignature(), parameterizedTypeName(StandardTypes.ARRAY, elementType.getTypeSignature()));
+
+        return new FunctionInfo(signature, getDescription(), isHidden(), methodHandle, isDeterministic(), false, ImmutableList.of(false));
+    }
+
+    public static void arrayWithUnknownType(MethodHandle compareMethodHandle, Type elementType, Slice array)
+    {
+    }
+
+    public static Long longArrayMinMax(MethodHandle compareMethodHandle, Type elementType, Slice array)
```

Comment:
Nice catch! Test cases works well with byte code generator after changing it to `long`